src/test/java/net/snowflake/client/jdbc/ConnectionLatestIT.java

```diff
@@ -843,13 +845,44 @@ public void testPrivateKeyInConnectionString() throws SQLException, IOException
           (int) ErrorCode.INVALID_OR_UNSUPPORTED_PRIVATE_KEY.getMessageCode(), e.getErrorCode());
     }
     connection.close();
+  }
 
-    // clean up
-    connection = getConnection();
-    statement = connection.createStatement();
-    statement.execute("use role accountadmin");
-    statement.execute(String.format("alter user %s unset rsa_public_key", testUser));
-    connection.close();
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testPrivateKeyInConnectionStringWithBouncyCastle() throws SQLException, IOException {
+    System.setProperty(ENABLE_BOUNCYCASTLE_PROVIDER_JVM, "true");
+    testPrivateKeyInConnectionString();
+  }
+
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testPbes2Support() throws SQLException, IOException {
+    // We're testing if we can decrypt a private key generated using OpenSSL v3.
+    // This was failing before with an invalid private key error before
+    // SNOW-896618 added the ability to decrypt private keys with
+    // -Dnet.snowflake.client.bouncyCastle=true
+
+    // The expectation is that this is going to fail (meaning the test passes when pbes2Supported is
+    // false) when we use the default JDK security providers without Bouncy Castle added.
+    // If the time comes when the JDK's default providers finally support PBES2 then this test will
+    // fail letting us know that we don't need the Bouncy Castle dependency anymore for private key
+    // decryption.
+    System.clearProperty(ENABLE_BOUNCYCASTLE_PROVIDER_JVM);
+    Boolean pbes2Supported = isPbes2KeySupported();
+    String failureMessage =
+        "The failure means that the JDK version can decrypt a private key generated by OpenSSL v3 and "
+            + "BouncyCastle shouldn't be needed anymore";
+    assertFalse(failureMessage, pbes2Supported);
+
+    // The expectation is that this is going to pass once we add Bouncy Castle in the list of
+    // providers
+    System.setProperty(ENABLE_BOUNCYCASTLE_PROVIDER_JVM, "true");
+    pbes2Supported = isPbes2KeySupported();
+    failureMessage =
+        "Bouncy Castle Provider should have been loaded with the -Dnet.snowflake.client.bouncyCastle=true "
+            + "JVM argument and this should have decrypted the private key generated by OpenSSL v3";
+    assertTrue(failureMessage, pbes2Supported);
+    System.clearProperty(ENABLE_BOUNCYCASTLE_PROVIDER_JVM);
```

Comment:
let's move the clearProperty to the method annotated with `@After` to be sure that it will be cleared